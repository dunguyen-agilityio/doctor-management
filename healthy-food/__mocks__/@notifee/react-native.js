export default {
  requestPermission: jest.fn(() => Promise.resolve(true)),
  displayNotification: jest.fn(() => Promise.resolve()),
  onForegroundEvent: jest.fn(),
  onBackgroundEvent: jest.fn(),
  createChannel: jest.fn(() => Promise.resolve('mock-channel')),
  getInitialNotification: jest.fn(() => Promise.resolve(null)),
};
