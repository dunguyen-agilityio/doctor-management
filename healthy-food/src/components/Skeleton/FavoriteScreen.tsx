import { StyleSheet, View } from 'react-native';

import { COLOR } from '@/theme';

import SearchInputSkeleton from './SearchInput';
import VerticalFoodListSkeleton from './VerticalFoodList';

const FavoriteScreenSkeleton = () => {
  return (
    <View style={styles.container}>
      <SearchInputSkeleton />
      <VerticalFoodListSkeleton />
    </View>
  );
};

export default FavoriteScreenSkeleton;

const styles = StyleSheet.create({
  container: {
    paddingHorizontal: 20,
    flex: 1,
    paddingTop: 55,
    backgroundColor: COLOR.WHITE,
    gap: 22,
  },
});
