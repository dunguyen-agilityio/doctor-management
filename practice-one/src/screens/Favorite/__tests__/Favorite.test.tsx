import React from 'react';

import { NavigationContainer, useFocusEffect } from '@react-navigation/native';

import { QueryClient, QueryClientProvider } from '@tanstack/react-query';
import { render } from '@testing-library/react-native';

import FavoriteScreen from '@/screens/Favorite';

jest.mock('@react-navigation/native', () => ({
  ...jest.requireActual('@react-navigation/native'),
  useFocusEffect: jest.fn(),
}));

describe('FavoriteScreen', () => {
  let queryClient: QueryClient;

  const renderWithProviders = () =>
    render(
      <QueryClientProvider client={queryClient}>
        <NavigationContainer>
          <FavoriteScreen />
        </NavigationContainer>
      </QueryClientProvider>,
    );

  beforeEach(() => {
    queryClient = new QueryClient({
      defaultOptions: { queries: { retry: false } },
    });
  });

  afterEach(() => {
    jest.clearAllMocks();
  });

  it('match snapshot', () => {
    const tree = renderWithProviders().toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('calls useFocusEffect when screen is focused', () => {
    renderWithProviders();

    expect(useFocusEffect).toHaveBeenCalled();
  });
});
