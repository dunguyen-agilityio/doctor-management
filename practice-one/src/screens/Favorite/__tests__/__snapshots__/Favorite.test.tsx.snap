// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FavoriteScreen match snapshot 1`] = `
<View
  style={
    {
      "backgroundColor": "#fff",
      "flex": 1,
      "paddingTop": 62,
    }
  }
>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "backgroundColor": "#FFF3F1",
        "borderRadius": 15,
        "marginHorizontal": 16,
        "marginTop": 14,
        "position": "relative",
        "zIndex": 1,
      }
    }
  >
    <TextInput
      onChangeText={[Function]}
      placeholder="Search for healthy food"
      placeholderTextColor="#FF8473"
      style={
        {
          "color": "#000000",
          "fontSize": 13,
          "fontWeight": "400",
          "paddingHorizontal": 60,
          "paddingVertical": 14,
        }
      }
      testID="search-input"
    />
    <View
      style={
        {
          "left": 21,
          "position": "absolute",
          "top": "50%",
          "transform": [
            {
              "translateY": -10,
            },
          ],
        }
      }
    >
      <Image
        source={
          {
            "testUri": "../../../src/assets/icons/iconsearch.png",
          }
        }
      />
    </View>
  </View>
  <View
    style={
      {
        "backgroundColor": "#fff",
        "flex": 1,
      }
    }
  >
    <View
      style={
        {
          "flex": 1,
          "paddingTop": 15,
        }
      }
    >
      <View
        style={
          {
            "backgroundColor": "#fff",
            "flex": 1,
            "gap": 18,
            "paddingBottom": 20,
            "paddingHorizontal": 20,
          }
        }
        testID="foods-list-container"
      >
        <RCTScrollView
          ListEmptyComponent={<NotFound />}
          contentContainerStyle={
            [
              {
                "gap": 18,
              },
              {
                "alignItems": "flex-start",
                "marginHorizontal": "auto",
                "width": 326,
              },
            ]
          }
          data={[]}
          getItem={[Function]}
          getItemCount={[Function]}
          keyExtractor={[Function]}
          onContentSizeChange={[Function]}
          onLayout={[Function]}
          onMomentumScrollBegin={[Function]}
          onMomentumScrollEnd={[Function]}
          onScroll={[Function]}
          onScrollBeginDrag={[Function]}
          onScrollEndDrag={[Function]}
          removeClippedSubviews={false}
          renderItem={[Function]}
          scrollEventThrottle={0.0001}
          showsHorizontalScrollIndicator={false}
          showsVerticalScrollIndicator={false}
          stickyHeaderIndices={[]}
          testID="foods-list"
          viewabilityConfigCallbackPairs={[]}
        >
          <View>
            <View
              style={
                {
                  "alignItems": "center",
                  "flex": 1,
                  "gap": 8,
                  "justifyContent": "center",
                  "width": "100%",
                }
              }
              testID="not-found-container"
            >
              <View
                style={
                  {
                    "alignItems": "center",
                    "gap": 24,
                  }
                }
                testID="not-found-header"
              >
                <Image
                  source={
                    {
                      "testUri": "../../../src/assets/images/not_found_2x.png",
                    }
                  }
                  style={
                    {
                      "height": 96,
                      "width": 96,
                    }
                  }
                  testID="not-found-image"
                />
                <Text
                  style={
                    {
                      "color": "#696969",
                      "fontSize": 22,
                      "fontWeight": "500",
                      "lineHeight": 31,
                      "marginTop": 24,
                    }
                  }
                >
                  No Foods Found
                </Text>
              </View>
              <Text
                style={
                  {
                    "color": "#5C5C5C",
                    "fontSize": 13,
                    "textAlign": "center",
                  }
                }
              >
                You don't save any food. Go ahead, search
and save your favorite food
              </Text>
            </View>
          </View>
        </RCTScrollView>
      </View>
    </View>
  </View>
  <View
    style={
      [
        {
          "alignItems": "center",
          "height": "100%",
          "justifyContent": "center",
          "position": "absolute",
          "width": "100%",
          "zIndex": 10,
        },
        {
          "marginTop": undefined,
        },
      ]
    }
    testID="loading-indicator"
  >
    <View
      collapsable={false}
      style={
        {
          "transform": [
            {
              "rotate": "0deg",
            },
          ],
        }
      }
    >
      <Image
        source={
          {
            "testUri": "../../../src/assets/images/logo.png",
          }
        }
        style={
          {
            "height": 122,
            "width": 122,
          }
        }
      />
    </View>
  </View>
</View>
`;
